### ISSUE TITLE
Syncing the agent on quicknode fails

### ISSUE THREAD

**Issue by compojoom** [beamer-bridge/beamer#1535]
#### System information
The agent doesn't seem to be able to sync with a quicknode RPC.

Traceback (most recent call last):
  File "/beamer/beamer/agent/chain.py", line 37, in wrapper
    return func(*args, **kwargs)
  File "/beamer/beamer/agent/chain.py", line 95, in _thread_func
    events.extend(fetcher.fetch())
  File "/beamer/beamer/agent/events.py", line 309, in fetch
    events = self._fetch_range(from_block, to_block)
  File "/beamer/beamer/agent/events.py", line 258, in _fetch_range
    logs = self._web3.eth.get_logs(params)
  File "/site-packages/web3/module.py", line 57, in caller
    result = w3.manager.request_blocking(method_str,
  File "/site-packages/web3/manager.py", line 197, in request_blocking
    response = self._make_request(method, params)
  File "/site-packages/web3/manager.py", line 150, in _make_request
    return request_func(method, params)
  File "/beamer/beamer/agent/middleware.py", line 275, in _rate_limiter
    return _rate_limiter_inner(method, params, make_request, w3, state)
  File "/beamer/beamer/agent/middleware.py", line 224, in _rate_limiter_inner
    rate_limited, response = _try_make_request(make_request, method, params)
  File "/beamer/beamer/agent/middleware.py", line 165, in _try_make_request
    raise exc
  File "/beamer/beamer/agent/middleware.py", line 161, in _try_make_request
    response = make_request(method, params)
  File "/beamer/beamer/agent/middleware.py", line 36, in middleware
    return make_request(method, params)
  File "/site-packages/web3/middleware/cache.py", line 140, in middleware
    return make_request(method, params)
  File "/site-packages/web3/middleware/signing.py", line 149, in middleware
    return make_request(method, params)
  File "/site-packages/web3/middleware/formatting.py", line 94, in middleware
    response = make_request(method, params)
  File "/site-packages/web3/middleware/gas_price_strategy.py", line 90, in middleware
    return make_request(method, params)
  File "/site-packages/web3/middleware/formatting.py", line 94, in middleware
    response = make_request(method, params)
  File "/site-packages/web3/middleware/attrdict.py", line 33, in middleware
    response = make_request(method, params)
  File "/site-packages/web3/middleware/formatting.py", line 94, in middleware
    response = make_request(method, params)
  File "/site-packages/web3/middleware/formatting.py", line 94, in middleware
    response = make_request(method, params)
  File "/site-packages/web3/middleware/formatting.py", line 94, in middleware
    response = make_request(method, params)
  File "/site-packages/web3/middleware/buffered_gas_estimate.py", line 40, in middleware
    return make_request(method, params)
  File "/site-packages/web3/middleware/formatting.py", line 94, in middleware
    response = make_request(method, params)
  File "/site-packages/web3/middleware/exception_retry_request.py", line 105, in middleware
    return make_request(method, params)
  File "/site-packages/web3/providers/rpc.py", line 88, in make_request
    raw_response = make_post_request(
  File "/site-packages/web3/_utils/request.py", line 103, in make_post_request
    response.raise_for_status()
  File "/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 413 Client Error: Request Entity Too Large for url: https://alpha-greatest-shard.optimism.discover.quiknode.pro/.../

#### Expected behavior
Agent should be able to sync

#### Actual behavior
Agent crashes.

#### Steps to reproduce the behavior
Try to run the agent with an rpc-url set to a quicknode RPC

**Comment by compojoom** [comment]
https://github.com/beamer-bridge/beamer/blob/80dccaf930c9bc33169e5cceadb47a931f83d17b/beamer/agent/events.py#L220

So basically the problem that when the request response is delivered very fast we increase the blocks we try to fetch. At one point quicknode responds with 413, because we are requesting a too big chunk, but since we were not catching HTTPErrors we were crashing. The fix is simple:

```
        # Some RPC providers return 413 Request Entity Too Large when the range is too big
        except requests.exceptions.HTTPError as exc:
            if exc.response.status_code == 413:
                old = self._blocks_to_fetch
                self._blocks_to_fetch = max(EventFetcher._MIN_BLOCKS, old // 2)
                self._log.debug(
                    "requests.exceptions.HTTPError 413: Request Entity Too Large, reducing number of blocks",
                    old=old,
                    new=self._blocks_to_fetch,
                )
                return None
            else:
                raise exc
```

@istankovic and I worked on refactoring the tests so that we can also test such exceptions in the event fetcher. 

I should have a PR for this tomorrow.

**Comment by istankovic** [comment]
Closed by #1545.
